Update custom-functions guide (#1000)
handsontable/hyperformula: docs/guide/custom-functions.md

# Custom functions

> Expand the function library of your application by adding custom functions.

Reviewer: ```suggestion
Expand the function library of your application, by adding custom functions.
```

Let's leave the comma here.

[[toc]]

## Add a custom function

> For example, let's create a function that returns the number of letters in the word "HyperFormula".

Reviewer: ```suggestion
As an example, let's create a function that returns the number of letters in the word "HyperFormula".
```

### 4. Implement your function's logic

> Add a method that implements your function's calculations in your function plugin. Your method needs to:

Reviewer: ```suggestion
In your function plugin, add a method that implements your function's calculations. Your method needs to:
```

Location comes first, action second.

You can set rules for your function's argument validation.

> In your function plugin, in the static `implementedFunctions` property, next to other options, add an array called `parameters`:

Reviewer: ```suggestion
In your function plugin, in the static `implementedFunctions` property, next to other options add an array called `parameters`:
```

#### Handling missing arguments

> The `defaultValue` and `optionalArg` options let you decide what happens when a user doesn't pass enough valid arguments to your custom function.

Reviewer: ```suggestion
Both the `defaultValue` and `optionalArg` options let you decide what happens when a user doesn't pass enough valid arguments to your custom function.
```
